packages/interface/src/utils/fetchFaucet.ts

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-console */
+import { ZKEdDSAEventTicketPCDClaim, ZKEdDSAEventTicketPCDPackage } from "@pcd/zk-eddsa-event-ticket-pcd";
+import { createWalletClient, http, isAddress, parseUnits } from "viem";
+import { mnemonicToAccount } from "viem/accounts";
+import { optimismSepolia } from "viem/chains";
+
+import { getDb } from "~/utils/db";
+
+import type { PCD } from "@pcd/pcd-types";
+import type { Groth16Proof } from "snarkjs";
+import type { IFetchFaucetReturn } from "~/utils/types";
+
+const localWalletClient = createWalletClient({
+  chain: optimismSepolia,
+  transport: http(),
+});
+
+const account = mnemonicToAccount(process.env.MNEMONIC!);
+const amountToSend = parseUnits(process.env.FAUCET_AMOUNT!, 16);
+const ZUPASS_SIGNER = "044e711fd3a1792a825aa896104da5276bbe710fd9b59dddea1aaf8d84535aaf";
+
+/**
+ * Handler for the faucet API.
+ * @param address - The address requests the funds.
+ * @param pcd - The pcd of that address.
+ * @returns - amount they get if succeeded, error message if failed.
+ */
+export async function fetchFaucet(
+  address: string,
+  pcd: PCD<ZKEdDSAEventTicketPCDClaim, Groth16Proof>,
+): Promise<IFetchFaucetReturn> {
+  /// Check if the address already claimed the faucet fund
+  const db = await getDb();
+
+  try {
+    // check if the address is valid
+    if (!isAddress(address)) {
+      throw new Error("Invalid address.");
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const alreadyUsed = await db.get("SELECT EXISTS(SELECT 1 FROM users WHERE pcd = ?)", pcd.id);
```

Comment:
Let's try something like this

```suggestion
    const alreadyUsed = await db.get("SELECT EXISTS(SELECT 1 FROM users WHERE pcd = ?)", pcd.id).then((res) => res as boolean);
```